app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -7,19 +8,33 @@ import createBlockTrackerInspectorMiddleware from 'eth-json-rpc-middleware/block
 import providerFromMiddleware from 'eth-json-rpc-middleware/providerFromMiddleware'
 import BlockTracker from 'eth-block-tracker'
 
+const inTest = process.env.IN_TEST === 'true'
+const blockTrackerOpts = inTest
+  ? { pollingInterval: 1000 }
+  : {}
+
 export default function createJsonRpcClient ({ rpcUrl, chainId }) {
   const fetchMiddleware = createFetchMiddleware({ rpcUrl })
   const blockProvider = providerFromMiddleware(fetchMiddleware)
-  const blockTracker = new BlockTracker({ provider: blockProvider })
+  const blockTracker = new BlockTracker({
+    ...blockTrackerOpts,
+    provider: blockProvider,
+  })
 
-  const networkMiddleware = mergeMiddleware([
+  const middlewares = [
     createChainIdMiddleware(chainId),
```

Comment:
It might be simpler to define a `testMiddlewares` array and spread that onto this list:

```suggestion
    ...testMiddlewares,
    createChainIdMiddleware(chainId),
```

This would simplify the diff but also keep all the test related code in one place—we could define `testMiddlewares` up alongside `inTest` and `blockTrackerOpts`.

Reply:
Done in: a2caf2e

Reply:
We can restore the single call to `mergeMiddleware` as well, that doesn't need to be split out anymore

Reply:
Fixed in: a665c2f